src/pages/RepoPage/FailedTestsTab/FailedTestsPage/FailedTestsPage.spec.tsx

```diff
@@ -0,0 +1,80 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { render, screen } from '@testing-library/react'
+import { graphql } from 'msw'
+import { setupServer } from 'msw/node'
+import { PropsWithChildren, Suspense } from 'react'
+import { MemoryRouter, Route } from 'react-router-dom'
+
+import FailedTestsPage from './FailedTestsPage'
+
+jest.mock('./SelectorSection/SelectorSection', () => () => 'Selector Section')
+jest.mock('./MetricsSection/MetricsSection', () => () => 'Metrics Section')
+jest.mock(
+  './FailedTestsTable/FailedTestsTable',
+  () => () => 'Failed Tests Table'
+)
+
+const server = setupServer()
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+      suspense: false,
+    },
+  },
+})
+
+const wrapper: (initialEntries?: string) => React.FC<PropsWithChildren> =
+  (initialEntries = '/gh/codecov/cool-repo/tests') =>
+  ({ children }) => (
+    <QueryClientProvider client={queryClient}>
+      <MemoryRouter initialEntries={[initialEntries]}>
+        <Route
+          path={[
+            '/:provider/:owner/:repo/tests',
+            '/:provider/:owner/:repo/tests/new',
+            '/:provider/:owner/:repo/tests/new/codecov-cli',
+            '/:provider/:owner/:repo/tests/:branch',
+          ]}
+          exact
+        >
+          <Suspense fallback={null}>{children}</Suspense>
+        </Route>
+      </MemoryRouter>
+    </QueryClientProvider>
+  )
+
+beforeAll(() => {
+  server.listen()
+})
+
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+describe('FailedTestsPage', () => {
+  function setup() {
+    server.use(
+      graphql.query('GetRepoOverview', (req, res, ctx) => {
+        return res(ctx.status(200), ctx.data({}))
+      })
+    )
+  }
+
+  it('renders sub-components', () => {
```

Comment:
It would be better to break these down into separate it blocks to improve debugging, but it's not a blocker since this file will be worked on later anyway

Reply:
I personally prefer this pattern, though I have seen us do the multiple renders and verbose testing that way.

My reasoning is that we have less renders overall, and if something fails on a certain line we know the exact column that fails anyway.

The perk with the verbose method though is if multiple columns fail you see them all at once